### PR TITLE
samples: subsys: fs: Support for CYW20829

### PR DESCRIPTION
**Pull request by sreeramIfx**
Use littlefs to store logs to flash memory

### CODE REVIEW

**Comment by de-nordic** [comment]
@sreeramIfx Compliance needs fixing.

**Comment by de-nordic** [comment]
@sreeramIfx Please fix the commit message and remove the - from the body.

**Review by de-nordic — APPROVED** [review]

**Comment by de-nordic** [comment]
@kartben @nashif Help with reviewing?

**Comment by kartben** [comment]
Not my area of expertise but is this really the recommend way to tap into, and potentially mess up with, the existing storage_partition? Most of the other overlays seem to declare a dedicated partition?

**Comment by de-nordic** [comment]
> Not my area of expertise but is this really the recommend way to tap into, and potentially mess up with, the existing storage_partition? Most of the other overlays seem to declare a dedicated partition?

TL;DR: it is fine, most default boards use default storage_partition.

This is sample, so it actually has entire board to mess up, so you actually mention quite important issue for me, but skip it for now.

If  you look through default configuration ,in sample.yaml, there is a lot of boards and they are not covered by any overlays. What really happens in code, at compile time there is attempt to identify whether there exists fstab definition for lfs1_part, and if not then storage_partition will be used - so for example nrf52840dk on default configuration with internal flash will run with storage_partition being used.
The overlays are provided for some configurations, where required, or in case of nrf52840dk spi/qspi overlays - to demonstrate how to use SPI with littlefs, so that people do not have to figure out by themselves.

Getting back at messing with storage_partition in the sample it may be ok, but still there is magic happening, because you need to know that default partition maps provided via DTS are something I have trouble with: the build system only honors them when MCUboot is enabled.
Unfortunately when you do not enabled MCUboot, the build system, for most boards, sees the entire flash available for image, so it can easily expand application into storage_partition and you would not even know it.

You probably understand now, that however you move the partition around in the end it is only a matter of time it will be in the wrong place.

**Review by kartben — APPROVED** [review]
